src/services/codefixes/fixUnusedIdentifier.ts

```diff
@@ -200,8 +200,14 @@ namespace ts.codefix {
 
     function tryDeleteParameter(changes: textChanges.ChangeTracker, sourceFile: SourceFile, p: ParameterDeclaration, checker: TypeChecker, sourceFiles: ReadonlyArray<SourceFile>, isFixAll: boolean): void {
         if (mayDeleteParameter(p, checker, isFixAll)) {
-            changes.delete(sourceFile, p);
-            deleteUnusedArguments(changes, sourceFile, p, sourceFiles, checker);
+            const privateModifier = findModifier(p, SyntaxKind.PrivateKeyword);
+            if (privateModifier) {
+                changes.deleteModifier(sourceFile, p.modifiers![0]);

```

Comment:
This needs to check if property is unused before this action can take place.

Reply:
@sheetalkamat  Can you give me a hint on how to do so? Do I have to walk the AST or are there already tools ready for that?

Reply:
@andy-ms to recommend better fix.

Reply:
I think we only provide an unused diagnostic if the property is unused. We won't provide a diagnostic when the parameter itself is unused because it is at least used to set the property.
I think the condition should be if `p.modifiers` exists at all, and it should delete all modifiers rather than just handling `private`; any other modifiers (`public`, `protected`, `readonly`) also make this a property.

Reply:
@andy-ms As mentioned in the related issue I'm adding the other modifiers.
The main question is if its okay to remove just the modifiers. There are cases where the property is flagged as unused, and if the modifiers are removed it may still be unused since the parameter is also unused.
Would a two-step fixing be okay? Or should it be in just one step?